internal/nexodus/proxyrule.go

```diff
@@ -0,0 +1,138 @@
+package nexodus
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+type ProxyType int
+
+const (
+	ProxyTypeEgress ProxyType = iota
+	ProxyTypeIngress
+)
+
+func (ruleType ProxyType) String() string {
+	switch ruleType {
+	case ProxyTypeEgress:
+		return "egress"
+	case ProxyTypeIngress:
+		return "ingress"
+	default:
+		panic(fmt.Sprintf("Invalid proxy rule type: %d", ruleType))
+	}
+}
+
+type ProxyProtocol string
+
+const (
+	proxyProtocolTCP ProxyProtocol = "tcp"
+	proxyProtocolUDP ProxyProtocol = "udp"
+)
+
+func parseProxyProtocol(protocol string) (ProxyProtocol, error) {
+	switch strings.ToLower(protocol) {
+	case "tcp":
+		return proxyProtocolTCP, nil
+	case "udp":
+		return proxyProtocolUDP, nil
+	default:
+		return "", fmt.Errorf("Invalid protocol (%s)", protocol)
+	}
+}
+
+type ProxyKey struct {
+	ruleType   ProxyType
+	protocol   ProxyProtocol
+	listenPort int
+}
+
+func (rule ProxyKey) String() string {
+	return fmt.Sprintf("%s:%s:%d", rule.ruleType, rule.protocol, rule.listenPort)
+}
+
+type ProxyRule struct {
+	ProxyKey
+	dest   HostPort
+	stored bool
+}
+
+type HostPort struct {
+	host string
+	port int
+}
+
+func (hp HostPort) String() string {
+	return fmt.Sprintf("%s:%d", hp.host, hp.port)
```

Comment:
need to use `JoinHostPort()` here. That will ensure that IPv6 addresses are handled correctly.

https://pkg.go.dev/net#JoinHostPort

Reply:
will do